flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/BaseExpressions.java

```diff
@@ -1407,6 +1409,44 @@ public OutType arrayUnion(InType array) {
                 unresolvedCall(ARRAY_UNION, toExpr(), objectToExpression(array)));
     }
 
+    /**
+     * Returns an array of the elements in the concat of array1 and array2, without duplicates.
+     *
+     * <p>If both of the array are null, the function will return null.
+     */
+    public OutType arrayConcat(InType... arrays) {
+        arrays = convertToArrays(arrays);
+        Expression[] args =
+                Stream.concat(
+                                Stream.of(toExpr()),
+                                Arrays.stream(arrays).map(ApiExpressionUtils::objectToExpression))
+                        .toArray(Expression[]::new);
+        return toApiSpecificExpression(unresolvedCall(ARRAY_CONCAT, args));
+    }
+
+    private InType[] convertToArrays(InType[] arrays) {
+        if (arrays == null || arrays.length < 1) {
+            throw new ValidationException("need at least two arrays");
```

Comment:
I didn't get it...
for the case of one array `arrays.length == 1` and it will not enter this `if`.
Also having just one array is ok, in that case we can just return it back.

Why are we talking about `"need at least two arrays"` ?